landingai/vision_pipeline.py

```diff
@@ -204,6 +210,66 @@ def save_image(self, filename_prefix: str, image_src: str = "") -> "FrameSet":
             c += 1
         return self
 
+    def save_video(
+        self,
+        video_file_path: str,
+        video_fps: Optional[int] = None,
+        video_length_sec: Optional[float] = None,
+        image_src: str = "",
+    ) -> "FrameSet":
+        """Save the FrameSet as an mp4 video file. The following example, shows to use save_video to save a clip from a live RTSP source.
+        ```python
+            video_len_sec=10
+            fps=4
+            img_src = NetworkedCamera(stream_url, fps=fps)
+            frs = FrameSet()
+            for i,frame in enumerate(img_src):
+                if i>=video_len_sec*fps: # Limit capture time
+                    break
+                frs.extend(frame)
+            frs.save_video("sample_images/test.mp4",video_fps=fps)
+        ```
+
+        Parameters
+        ----------
+        video_file_path : str
+            Path and filename with extension of the video file
+        video_fps : Optional[int]
+            The number of frames per second for the output video file.
+            Either the `video_fps` or `video_length_sec` should be provided to assemble the video. if none of the two are provided, the method will try to set a "reasonable" value.
+        video_length_sec : Optional[float]
+            The total number of seconds for the output video file.
+            Either the `video_fps` or `video_length_sec` should be provided to assemble the video. if none of the two are provided, the method will try to set a "reasonable" value.
+        image_src : str, optional
+            if empty the source image will be used. Otherwise the image will be selected from `other_images`
+        """
+        if not video_file_path.lower().endswith(".mp4"):
+            raise NotImplementedError("Only .mp4 is supported")
+        total_frames = len(self.frames)
+        if total_frames == 0:
+            return self
+
+        if video_fps is not None and video_length_sec is not None:
+            raise ValueError(
+                "The 'video_fps' and 'video_length_sec' arguments cannot be set at the same time"
+            )
+
+        # Try to tune FPS based on parameters or pick a reasonable number. The goal is to produce a video that last a a couple of seconds even when there are few frames. OpenCV will silently fail and not create a file if the resulting fps is less than 1
+        if video_length_sec is not None and video_length_sec <= total_frames:
```

Comment:
got it, if FPS can't be lower than 1, how about the API automatically caps it at 1 fps? see below

```suggestion
        if video_length_sec > total_frames:
              video_length_sec = total_frames        
        if video_length_sec is not None and video_length_sec <= total_frames:
```

it is worth a documentation to avoid confusion and clarify the behavior.

Reply:
This is already happening in line 260.... I try to make the default FPS 2 (for faster video) but if there is a single frame then FPS = 1....

Reply:
i think line 260 doesn't cover the case when both `video_length_sec` and `video_fps` is not None.
in addition, the current behavior is a bit confusing too. let's discuss it in our standup.

Reply:
Added to dicsussion list